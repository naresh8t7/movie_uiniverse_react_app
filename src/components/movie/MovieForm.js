import React from 'react';
import { Field, reduxForm } from 'redux-form';
import FieldInput from '../common/FieldInput';


export const MovieForm = ({ handleSubmit, pristine, reset, submitting, heading, handleSave, handleCancel }) => {
    return (
        <form onSubmit={handleSubmit(handleSave)}>
            <h1>{heading}</h1>

            <Field
                type="text"
                name="title"
                label="Title"
                placeholder="Title of the movie"
                component={FieldInput}
            />


            <Field
                type="number"
                name="year"
                label="year"
                placeholder="Year of the movie"
                component={FieldInput}
            />

            <div>
                <button type="submit" disabled={submitting} className="btn btn-primary"><i className="fa fa-paper-plane-o" aria-hidden="true" /> Submit</button>

                {heading === 'Add Movie' && <button type="button" disabled={pristine || submitting} onClick={reset} className="btn btn-default btn-space">Reset</button>}

                <button type="button" className="btn btn-default btn-space" onClick={handleCancel}>Cancel</button>
            </div>
        </form>
    );
};





const validate = values => {
    const errors = {};

    if (!values.title) {
        errors.title = 'Required Title';
    }

    if (!values.year) {
        errors.year = 'Required Year';
    }
    return errors;
};



export default reduxForm({
    form: 'MovieForm',
    validate
})(MovieForm);
