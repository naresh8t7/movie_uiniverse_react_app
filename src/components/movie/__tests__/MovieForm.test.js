import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {MovieForm} from '../MovieForm';



function setup(heading) {
    const props = {
        handleSubmit: jest.fn(),
        pristine: true,
        reset: jest.fn(),
        submitting: false,
        heading: heading,
        handleSave: jest.fn(),
        handleCancel: jest.fn(),
    };

    return shallow(<MovieForm {...props}/>);
}


describe('MovieForm.test.js', () => {

    it('renders without crashing', () => {
        const wrapper = setup('Add');
        expect(wrapper.length).toEqual(1);
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });


    it('renders form and display "Add Movie" in h1', () => {
        const wrapper = setup('Add Movie');
        expect(wrapper.find('h1').text()).toEqual('Add Movie');
    });


    it('displays "Edit Movie" in h1', () => {
        const wrapper = setup('Edit Movie');
        expect(wrapper.find('h1').text()).toEqual('Edit Movie');
    });

    
});

