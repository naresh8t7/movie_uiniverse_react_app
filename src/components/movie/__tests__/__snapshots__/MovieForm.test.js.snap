// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieForm.test.js renders without crashing 1`] = `
<form>
  <h1>
    Add
  </h1>
  <Field
    component={[Function]}
    label="Title"
    name="title"
    placeholder="Title of the movie"
    type="text"
  />
  <Field
    component={[Function]}
    label="year"
    name="year"
    placeholder="Year of the movie"
    type="number"
  />
  <div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      <i
        aria-hidden="true"
        className="fa fa-paper-plane-o"
      />
       Submit
    </button>
    <button
      className="btn btn-default btn-space"
      onClick={[MockFunction]}
      type="button"
    >
      Cancel
    </button>
  </div>
</form>
`;
